minicluster/src/test/java/org/apache/accumulo/minicluster/MiniAccumuloClusterTest.java

```diff
@@ -210,6 +210,11 @@ public void testPerTableClasspath() throws Exception {
     conn.instanceOperations().setProperty(VFS_CONTEXT_CLASSPATH_PROPERTY.getKey() + "cx1",
         jarFile.toURI().toString());
 
+    // Batchwriter is intermittently failing with a constraint violation because the TabletServer
+    // is not seeing the property changes above before the BatchWriter below closes and a
+    // MutationsRejectedException is being thrown.
+    Thread.sleep(5000);
```

Comment:
Would something like this work instead of just the sleep to make sure the property is set?
```suggestion
    String classpathKey = VFS_CONTEXT_CLASSPATH_PROPERTY.getKey() + "cx1";
    String classpathValue = jarFile.toURI().toString();
    conn.instanceOperations().setProperty(classpathKey, classpathValue);

    Map<String,String> systemConfiguration = conn.instanceOperations().getSystemConfiguration();

    while (!systemConfiguration.containsKey(classpathKey)
        || !systemConfiguration.get(classpathKey).equals(classpathValue)) {

      log.info("Expected {} but instead found {}. Sleeping for 1000ms", classpathValue,
          systemConfiguration.get(classpathKey));
      Thread.sleep(1000);
      systemConfiguration = conn.instanceOperations().getSystemConfiguration();
    }
```
I haven't been able to replicate the test failure but the test passes with the changes above.

Reply:
It's likely that the test thread will see the configuration change, it's the fact that the TabletServer does *not* see the change before the BatchWriter is used. I might have to move this one test to its own class.

Reply:
Checking if the property is set will only verify that the change has propagated locally. Each process will have its own cache and sets of watchers on the property node.  So checking if the property change has propagated in one process does not provide any guarantees that all of the other processes have synced.